src/relay/transforms/pattern_util.h

```diff
@@ -374,7 +374,7 @@ inline bool IsEqualScalar(const Expr& a, const Expr& b) {
  * \param i element index
  * \return Converted scalar value.
  */
-static inline double ToScalar(const runtime::NDArray& array, size_t i = 0) {
+static inline long double ToScalar(const runtime::NDArray& array, size_t i = 0) {
```

Comment:
why double is not sufficient?

Reply:
Because a double only has 52 bits of mantissa, it can't store the full precision of an int64_t.

Reply:
So we get rounding errors if we pass in large int64_t values

Reply:
long double has 63 bits of mantissa and 1 bit of sign, just like int64